src/test/resources/epub-indexes/indexes-package-document.feature

```diff
@@ -0,0 +1,53 @@
+Feature: EPUB Indexes Package Document
+  
+  Checks conformance to rules for Package Documents defined in the
+  EPUB Indexes specification:
+    
+    http://idpf.org/epub/idx/
+  
+  This feature file contains tests for EPUBCheck running in `opf` mode to check
+  single Package Documents (`.opf` files).
+  
+  Note: 
+  - Tests related to EPUB Indexes package rules in a full EPUB publication
+    are defined in the `indexes-package.feature` feature file.
+  - Tests related to general conformance of Package Documents are defined
+    in the `epub3/package-document.feature` feature file.
+
+  Background: 
+    Given EPUB test files located at '/epub-indexes/files/package-document/'
+    And EPUBCheck configured to check a Package Document
+
+  ##  2.3.2.2 Multi-File Index(es) and the collection Element
+
+  Scenario: An index collection is reported as valid
+    When checking file 'index-collection-valid.opf'
+    Then no errors or warnings are reported
+    
+  Scenario: An index collection may contain index-group child collections
+    When checking file 'index-collection-index-group-valid.opf'
+    Then no errors or warnings are reported
+    
+  Scenario: An index collection must only contain links to XHTML Content Documents
+    When checking file 'index-collection-resource-not-xhtml-error.opf'
+    Then error OPF-071 is reported
+    And the message contains "Index collections must only contain resources pointing to XHTML Content Documents"
```

Comment:
Is adding messages into the tests going to make their maintenance more time-consuming going forward? To change one, you'd now have to update the message itself plus every feature it might be raised in.

And more a question, but how do you decide when to include a message matching case?

Reply:
Yeah I've been wondering about that TBH.

Regarding the brittleness of including the messages in the tests, I think it's generally OK as long as we don't have too many tests quoting the same message, which shouldn't be the case anyway. (ideally, we should only have a very few tests about a specific message). Also it's an easy refactoring with global search and replace. So I'm not too worried.

I tried to add messages when it helped with the understanding of the test (but to be fair, it does seem redundant in the snippet quoted above 😅).

The one case where the message matching test is absolutely useful is for the few generic errors that are associated to many different messages, like RSC-005 and RSC-017.

Reply:
>The one case where the message matching test is absolutely useful is for the few generic errors that are associated to many different messages, like RSC-005 and RSC-017.

Ya, I thought of this as I was falling asleep last night. (seriously, epubcheck issues will knock you right out! ;P )

Given all the different messages that might come out for RSC-005, for example, it definitely makes sense to be specific. There's always a chance that one error could be replaced by an entirely different one and the build process wouldn't care as the count would be the same.